nncf/common/pruning/operations.py

```diff
@@ -154,6 +154,18 @@ def mask_propagation(cls, node: NNCFNode, graph: NNCFGraph,
         node.data['output_mask'] = output_mask
 
 
+class EmbeddingPruningOp(BasePruningOp):
+    @classmethod
+    def accept_pruned_input(cls, node: NNCFNode) -> bool:
+        return False
+
+    @classmethod
+    def mask_propagation(cls, node: NNCFNode, graph: NNCFGraph,
+                         tensor_processor: Type[NNCFPruningBaseTensorProcessor]) -> None:
+        output_mask = node.data.get('output_mask', None)
+        node.data['output_mask'] = output_mask
+
+
```

Comment:
Is it used somewhere? The logic here allows to produce masks, but it's never used in the pruning algo.

Reply:
It's used here: https://github.com/openvinotoolkit/nncf/blob/develop/nncf/experimental/torch/nas/bootstrapNAS/elasticity/elastic_width.py#L625
Embedding became a prunable type and part of `PT_PRUNING_OPERATOR_METATYPES`, we set elastic width encoded in the `output_mask` and propagate them. 
```python
algo = MaskPropagationAlgorithm(
            self._propagation_graph, PT_PRUNING_OPERATOR_METATYPES, PTNNCFPruningTensorProcessor)
        algo.mask_propagation()
```

Reply:
I'm only concerned that filter pruning user doesn't expect embedding layer could possibly produce masks. It works now and doesn't introduce any side effects for FilterPruning as far as I understand(because embedding layer isn't updated with preop, and model analysis doesn't consider embedding as prunable), but in general this PruningOp looks unfinished from Filter Pruning algorithm side.
This change looks implicit, to enable elastic embedding for NAS algo but shown as common. I would be concerned if I would see this